mesonbuild/dependencies/cmake.py

```diff
@@ -654,3 +654,27 @@ def __call__(self, name: str, env: Environment, kwargs: T.Dict[str, T.Any], lang
     @staticmethod
     def log_tried() -> str:
         return CMakeDependency.log_tried()
+
+
+def sort_link_args(args: T.List[str]) -> T.List[str]:
+    itr = iter(args)
+    result = set()
```

Comment:
You'll need to help the type checker out here:
```suggestion
    result: T.Set[T.Union[T.Tuple[str], T.Tuple[str, str]]] = set()
```